lndr-backend/src/Lndr/Handler/Friend.hs

```diff
@@ -79,8 +81,23 @@ userHandler Nothing Nothing = throwError (err400 {errBody = "No identifying info
 addFriendsHandler :: Address -> [Address] -> LndrHandler NoContent
 addFriendsHandler address friendAddresses = do
     -- TODO verify signature
-    pool <- asks dbConnectionPool
+    (ServerState pool configTVar loggerSet) <- ask
+    config <- liftIO $ readTVarIO configTVar
     liftIO . withResource pool $ Db.addFriends ((address,) <$> friendAddresses)
+    
+    let attemptToNotify friendAddress = do
+            pendingRequests <- liftIO . withResource pool $ Db.lookupPendingRequest address friendAddress
+            when (pendingRequests > 0) $ do
+                pushDataM <- liftIO . withResource pool $ Db.lookupPushDatumByAddress friendAddress
+                nicknameM <- liftIO . withResource pool $ Db.lookupNick address
```

Comment:
This can be pushed up in scope; currently, the query is performed each time `attemptToNotify` is invoked. Instead, it can be performed once, directly in the `addFriendsHandler` scope, and referenced.